internal/runners/projects/move.go

```diff
@@ -0,0 +1,126 @@
+package projects
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/prompt"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/ActiveState/cli/pkg/projectfile"
+)
+
+type MoveParams struct {
+	Namespace *project.Namespaced
+	NewOwner  string
+}
+
+type Move struct {
+	auth   *authentication.Auth
+	out    output.Outputer
+	prompt prompt.Prompter
+	config configGetter
+}
+
+func NewMove(prime primeable) *Move {
+	return &Move{
+		auth:   prime.Auth(),
+		out:    prime.Output(),
+		prompt: prime.Prompt(),
+		config: prime.Config(),
+	}
+}
+
+func NewMoveParams() *MoveParams {
+	return &MoveParams{Namespace: &project.Namespaced{}}
+}
+
+func (m *Move) Run(params *MoveParams) error {
+	if !m.auth.Authenticated() {
+		return locale.NewInputError("err_project_move_auth", "In order to move your project you need to be authenticated. Please run '[ACTIONABLE]state auth[/RESET]' to authenticate.")
+	}
+
+	defaultChoice := !m.out.Config().Interactive
+	move, err := m.prompt.Confirm("", locale.Tl("move_prompt",
+		"You are about to move the project [NOTICE]{{.V0}}[/RESET] to the organization [NOTICE]{{.V1}}[/RESET].\nContinue?",
+		params.Namespace.String(), params.NewOwner), &defaultChoice)
+	if err != nil {
+		return locale.WrapError(err, "err_move_prompt", "Could not prompt for move confirmation")
+	}
+
+	if !move {
+		m.out.Print(locale.Tl("move_cancelled", "Project move aborted by user"))
+		return nil
+	}
+
+	if err = model.MoveProject(params.Namespace.Owner, params.Namespace.Project, params.NewOwner, m.auth); err != nil {
+		return locale.WrapError(err, "err_move_project", "Could not move project")
+	}
+
+	if err = m.updateLocalCheckouts(params); err != nil {
+		return locale.WrapError(err, "err_edit_local_checkouts")
+	}
+
+	if err = m.updateProjectMapping(params); err != nil {
+		return locale.WrapError(err, "err_edit_project_mapping")
+	}
+
+	m.out.Notice(locale.Tl("move_success",
+		"Project [NOTICE]{{.V0}}[/RESET] successfully moved to the [NOTICE]{{.V1}}[/RESET] organization",
+		params.Namespace.Project, params.NewOwner))
+
+	return nil
+}
+
+func (m *Move) updateLocalCheckouts(params *MoveParams) error {
```

Comment:
Is there any appetite for unifying this and the edit `updateLocalCheckouts` function somewhere like the `projectfile` package?

Reply:
I was hoping to do so, but the logic in the loops is slightly different, and the input parameters are different. Trying to shoehorn them into some sort of union didn't seem worth it and would probably cause more complexity than it's worth.